training/sklearn/structured/custom_routines/README.md

```diff
@@ -0,0 +1,188 @@
+  good idea to try and train the model locally for debugging, before submitting it to AI Platform.
+  * [train-cloud.sh](./scripts/train-cloud.sh) submits a training job to AI Platform.
+* [setup.py](./setup.py): containing all the required Python packages for this tutorial.
+
+
+We recommend that you follow the same structure for your own work. In most cases, you only need to 
+modify:
+
+ - `metadata.py`
+ - `model.py`
+ 
+ and leave the other python files untouched.
+
+## Running the Sample
+
+After you go over the steps in the prerequisites section, you are ready to run this sample.
+Here are the steps you need to take:
+
+1. _[Optional]_ Train the model locally. Run:
+ 
+```bash
+source ./scripts/train-local.sh
+``` 
+
+as many times as you like (This has no effect on your cloud usage). If successful, this script should
+create a new model as `trained/structured-taxi/model.joblib`, which means you may now submit a
+training job to AI Platform.
+
+2. Submit a training job to AI Platform. Run: 
+
+```bash
+source ./scripts/train-cloud.sh
+``` 
+This will create a training job on AI Platform and displays some instructions on how to track the job progress.
+At the end of a successful training job, it will upload the trained model object to a GCS
+bucket and sets `$MODEL_DIR` environment variable to the parent directory of all the generated models.
+
+It will also package up the custom routine and upload it to the bucket and 
+set the environment variable `CUSTOM_ROUTINE_PATH` which points to it.
+`CUSTOM_ROUTINE_PATH` will later be used during the deployment of the model.
+
+### Monitoring
+Once the training starts and the models are generated, you may view the training job in
+the [AI Platform page](https://pantheon.corp.google.com/mlengine/jobs). If you click on the 
+corresponding training job, you will be able to view the chosen hyperparamters, along with the
+metric scores for each model. All the generated model objects will be stored on GCS. 
+
+## Explaining Key Elements
+
+In this section, we'll highlight the main elements of this sample.
+
+### [model.py](trainer/model.py)
+
+In this sample, we simply create an instance of `RandomForestClassifier` estimator and return it.
+
+### [metadata.py](trainer/metadata.py)
+
+We define which features should be used for training. We also define what the target is.
+
+### [my_pipeline.py](trainer/my_pipeline.py)
+
+This file contains our custom routines. The code in this file is required for making predictions.
+We will package and ship this, along with our trained model
+when we deploy our model to AI Platform to make predictions.
+
+### [train-local.sh](./scripts/train-local.sh)
+
+The command to run the training job locally is this:
+
+```bash
+gcloud ai-platform local train \
+        --module-name=trainer.task \
+        --package-path=${PACKAGE_PATH} \
+        --job-dir=${MODEL_DIR} \
+        -- \
+        --log-level DEBUG \
+        --input=${TAXI_TRAIN_SMALL}
```

Comment:
Is this GCS_TAXI_TRAIN_SMALL ?

Reply:
Not for local training. For that we actually do use TAXI_TRAIN_SMALL (which is set after running download-taxi.sh).